test/functional/ex_cmds/source_spec.lua

```diff
@@ -0,0 +1,34 @@
+local helpers = require('test.functional.helpers')(after_each)
+local command = helpers.command
+local insert = helpers.insert
+local wait = helpers.wait
+local eq = helpers.eq
+local clear = helpers.clear
+local meths = helpers.meths
+local feed = helpers.feed
+local feed_command = helpers.feed_command
+
+describe(':source', function()
+  before_each(function()
+    clear()
+  end)
+
+  it('source current buffer', function()
+    insert('let a = 2')
+    wait()
+    command('source')
+    eq('2', meths.exec('echo a', true))
+  end)
+
+  it('source selection in current buffer', function()
+    insert(
+      'let a = 2\n'..
+      'let a = 3\n'..
+      'let a = 4\n')
+    wait()
+    -- Source the 2nd line only
+    feed('ggjV')
+    feed_command(':source')
+    eq('3', meths.exec('echo a', true))
```

Comment:
None of the current tests have tried sourcing multiple lines, so I would add that. Try sourcing lines 2-$ and see if `a == 4`

Reply:
Done.